app/controllers/student_lessons_controller.rb

```diff
@@ -47,4 +53,8 @@ def filled_grades_attributes
   def grades_attributes
     params.require(:student).permit(grades_attributes: %i[id skill grade_descriptor_id])[:grades_attributes].values
   end
+
+  def lesson_from_param
+    Lesson.includes(subject: [skills: [:grade_descriptors]]).find params[:lesson_id]
```

Comment:
`[skills:` that syntax looks a little off to me, perhaps just new - that creates a hash inside an array?

Reply:
Should be the same as:
```ruby
Lesson.includes(subject: [{ skills: [:grade_descriptors] }])
```

You do have a point, it looks syntactically off. I'll add braces to make it clearer